app/docs/md/patterns/event-listeners.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Event Listeners
+---
+
+An event is a signal that something has happened on your page. The browser notifies you so you can react to them.
+
+## What are events?
+
+From [MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#what_is_an_event)
+
+> Events are things that happen in the system you are programming — the system produces (or "fires") a signal of some kind when an event occurs, and provides a mechanism by which an action can be automatically taken (that is, some code running) when the event occurs. Events are fired inside the browser window, and tend to be attached to a specific item that resides in it. This might be a single element, a set of elements, the HTML document loaded in the current tab, or the entire browser window. There are many different types of events that can occur.
+
+For example:
+
+- A user submits a form
+- A user clicks on a button
+- A web page finishes loading
+- etc.
+
+**[Learn more about events](https://developer.mozilla.org/en-US/docs/Web/Events)**
+
+## Adding a click event listener
+
+Suppose we have a simple web component called `custom-button` that we want to add a click event listener to.
+
+<doc-code filename="app/components/custom-button.mjs">
+
+```javascript
+import CustomElement from '@enhance/custom-element'
+
+export default class CustomButton extends CustomElement {
+  render ({ html, state }) {
+    return html`
+        <style>
+            button {
+                border: 1px solid white;
+                padding: 2px;
+            }
+        </style>
+        <button>Click me!</button>
+    `
+  }
+}
+
+customElements.define('custom-button', CustomButton)
+```
+
+</doc-code>
+
+As a best practice, event listeners for web components should be added inside the `connectedCallback` function, since this method runs when the element is added to the DOM.
```

Comment:
IIRC, @kristoferjoseph just ran into an issue w/ DSD where event listeners need to be registered in the constructor instead? May need to clarify here in case.

Reply:
With DSD and SSR a user does have access to the entire template generated DOM but `connectedCallback` is the correct approach that to document since it works for both but only DSD and SSR support adding event listeners in the constructor.

The main issue being that even if the user is able to attach listeners in the constructor after initial render they may try to use the same class client side without SSR or DSD and be confused as to why it doesn’t work.

Reply:
Ahh, so I had it backwards? Coooool 😄